nifi-nar-bundles/nifi-hive-bundle/nifi-hive-processors/src/main/java/org/apache/nifi/processors/hive/SelectHiveQL.java

```diff
@@ -290,6 +292,9 @@ public void process(final OutputStream out) throws IOException {
                             }
                         }
 
+                        // set query timeout
+                        st.setQueryTimeout(queryTimeout);
```

Comment:
Nothing to be done here necessarily, but just wanted to mention that the success of this method is dependent on the customValidate() preventing the processor from running if the Hive JDBC driver does not support setQueryTimeout.

Reply:
I get the point of the customValidate but I'd keep its check but mark it as valid whether the timeout method is supported or not.  YOu can set some processor instance boolean to advise whether it is supported then in this call/check set the timeout if it is and ignore it if not. Otherwise we're requiring them to use a version of hive which supports it and that seems a little heavy handed.  It is probably a good idea to do the current validation logic in some other lifecycle call like 'onAdded' or something, warn if timeouts not supported and point out hanging threads are possible, and keep going.

Reply:
disregard what i said.  matt pointed out that the code checks IF a timeout was set and is non-zero.  Based on that I think the way it was implemented is awesome.